### PR TITLE
Optimize contest results validation and submissions by problem query

### PR DESCRIPTION
**Pull request by gogo4ds**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1645
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1643

**Summary of the changes made**:
1. Optimized validation logic
2. Code style
3. Optimize submissions by problem query

### CODE REVIEW

File: Services/UI/OJS.Services.Ui.Business/Validations/Implementations/Contests/ContestResultsValidationService.cs [hunk 1]
```diff
@@ -1,87 +1,44 @@
...
 
-        var contestActivity = await this.activityService.GetContestActivity(contest!.Map<ContestForActivityServiceModel>());
-
-        var contestHasStartDateAndEndTimeIsBeforeNow =
-            contest.StartTime.HasValue &&
-            contest.EndTime.HasValue &&
-            contest.EndTime <= this.datesService.GetUtcNow();
-
-        if (isOfficial && !contestHasStartDateAndEndTimeIsBeforeNow && await this.participantsData
-                .GetByContestByUserAndByIsOfficial(contest.Id, user!.Id, isOfficial) == null)
```
Comment (deyordanov) [inline Services/UI/OJS.Services.Ui.Business/Validations/Implementations/Contests/ContestResultsValidationService.cs:79]: Why did you remove this validation?

**Review by deyordanov — CHANGES_REQUESTED** [review]

File: Services/UI/OJS.Services.Ui.Business/Validations/Implementations/Contests/ContestResultsValidationService.cs (hunk 1 above)
Comment (gogo4ds) [reply Services/UI/OJS.Services.Ui.Business/Validations/Implementations/Contests/ContestResultsValidationService.cs:79]: It's inlcuded in the contestActivity result.

**Review by deyordanov — APPROVED** [review]
